backend/services/studentService.js

```diff
@@ -28,8 +29,21 @@ module.exports = {
         }
         return studentDAO.setStudentSkills(studentId, skills)    
     },
-    // getStudentsBySkills: (skills) => {
-    //     let skillIds = skills.map(skill => skill.skill_id);
-    //     return studentDAO.getStudentsBySkills(skillIds);
-    // },
+
+    getStudentsBySkills: async (skills) => {
+        let skillIds = skills.map(skill => skill.skill_id);
+        return await studentDAO.getStudentsBySkills(skillIds);
+    },
+
+    getStudentProfile: async (id) => {
+        let profile = await studentDAO.getStudentById(id);
+        profile = profile[0];
```

Comment:
let's put this line where we get the first element of the resulted query in the DAO method, in "getStudentById". The service should not care that Knex returns us all results in an array, we should send to this directly the single result that we are interested in. When you add this in "getStudentById" you have to make sure you make the method async as well

Reply:
I changed it to the version you suggested and made it async